Sample Applications/DataBindingDemo/Styles.xaml

```diff
@@ -8,36 +8,79 @@
-        <Setter Property="Foreground" Value="DimGray" />
+        <Setter Property="Label.FontWeight" Value="Bold" />
         <Setter Property="VerticalAlignment" Value="Center" />
         <Setter Property="HorizontalAlignment" Value="Right" />
     </Style>
 
+    <Style x:Key="SmallTitleStyleOnFocus" TargetType="TextBlock" BasedOn="{StaticResource SmallTitleStyle}">
+        <Style.Triggers>
+            <!-- High contrast support -->
+            <DataTrigger Binding="{Binding Path=(SystemParameters.HighContrast)}" Value="true" >
+                <Setter Property="Foreground" Value="{DynamicResource {x:Static SystemColors.HighlightTextBrushKey}}" />
+            </DataTrigger>
+        </Style.Triggers>
+    </Style>
+
+    <Style x:Key="SmallTitleStyleOnFocusLost" TargetType="TextBlock" BasedOn="{StaticResource SmallTitleStyle}">
+        <Style.Triggers>
+            <!-- High contrast support -->
+            <DataTrigger Binding="{Binding Path=(SystemParameters.HighContrast)}" Value="true" >
+                <Setter Property="Foreground" Value="{DynamicResource {x:Static SystemColors.WindowTextBrushKey}}" />
+            </DataTrigger>
+        </Style.Triggers>
+    </Style>
+
     <Style x:Key="TextStyleTextBlock" TargetType="TextBlock">
-        <Setter Property="Foreground" Value="#333333" />
+        <Setter Property="Label.FontWeight" Value="Normal" />
+    </Style>
+
+    <Style x:Key="TextStyleTextBlockOnFocus" TargetType="TextBlock" BasedOn="{StaticResource TextStyleTextBlock}">
+        <Style.Triggers>
+            <!-- High contrast support -->
+            <DataTrigger Binding="{Binding Path=(SystemParameters.HighContrast)}" Value="true" >
+                <Setter Property="Foreground" Value="{DynamicResource {x:Static SystemColors.HighlightTextBrushKey}}" />
+            </DataTrigger>
+        </Style.Triggers>
     </Style>
 
     <Style x:Key="TextStyleTextBox" TargetType="TextBox">
         <Setter Property="Foreground" Value="#333333" />
         <Setter Property="MaxLength" Value="40" />
         <Setter Property="Width" Value="392" />
         <Style.Triggers>
+            <!-- High contrast support -->
+            <DataTrigger Binding="{Binding Path=(SystemParameters.HighContrast)}" Value="true" >
+                <Setter Property="Foreground" Value="{DynamicResource {x:Static SystemColors.GrayTextBrushKey}}" />
+            </DataTrigger>
+            <MultiDataTrigger>
+                <MultiDataTrigger.Conditions>
+                    <Condition Binding="{Binding Path=(SystemParameters.HighContrast)}" Value="true" />
+                    <Condition Binding="{Binding Path=MouseOver, RelativeSource={RelativeSource Self}}" Value="true" />
+                </MultiDataTrigger.Conditions>
+                <MultiDataTrigger.Setters>
+                    <Setter Property="TextBlock.Foreground" Value="{DynamicResource {x:Static SystemColors.HighlightTextBrushKey}}" />
```

Comment:
Are you sure about this color choice?  In some themes, highlight text is the same color as the window background.  If you are not also setting the control background to HighlightBrushKey, you can end up with invisible text.

Reply:
Similar comments for all uses of HighlightTextBrush where it is the sole brush change.

Reply:
I have made some changes